source/includes/fundamentals/code-snippets/enterprise-auth.rs

```diff
@@ -17,5 +25,86 @@ async fn main() -> mongodb::error::Result<()> {
     let client = Client::with_options(client_options)?;
     // end-ldap
 
+    // start-azure-imds
+    let credential = Credential::builder()
+        .username("<username>".to_owned())
+        .mechanism(AuthMechanism::MongoDbOidc)
+        .mechanism_properties(
+            doc! { "ENVIRONMENT": "azure", "TOKEN_RESOURCE": "<audience>" }
+        )
+        .build();
+    
+    client_options.credential = Some(credential);
+    let client = Client::with_options(client_options)?;
+    let res = client
+        .database("test")
+        .collection::<Document>("test")
+        .find_one(doc! {})
+        .await?;
+    // end-azure-imds
+
+    // start-gcp-imds
+    let credential = Credential::builder()
+        .mechanism(AuthMechanism::MongoDbOidc)
+        .mechanism_properties(
+            doc! { "ENVIRONMENT": "gcp", "TOKEN_RESOURCE": "<audience>" }
+        )
+        .build();
+    
+    client_options.credential = Some(credential);
+    let client = Client::with_options(client_options)?;
+    let res = client
+        .database("test")
+        .collection::<Document>("test")
+        .find_one(doc! {})
+        .await?;
+    // end-gcp-imds
+
+    // start-custom-callback-machine
+    let credential = Credential::builder()
+    .mechanism(AuthMechanism::MongoDbOidc)
+    .oidc_callback(oidc::Callback::machine(move |_| {
+        async move {
+            let token_file_path = std::env::var("AWS_WEB_IDENTITY_TOKEN_FILE").map_err(mongodb::error::Error::custom)?;
+            let access_token = tokio::fs::read_to_string(token_file_path).await?;
+            Ok(IdpServerResponse::builder().access_token(access_token).build())
+        }
+        .boxed()
+    }))
+    .build()
+    .into();
+
+    client_options.credential = Some(credential);
+    let client = Client::with_options(client_options)?;
+
+    let res = client
+    .database("test")
+    .collection::<Document>("test")
+    .find_one(doc! {})
+    .await?;
+    // end-custom-callback-machine
+
+    // start-custom-callback-user
+    let callback = Callback::human(move |context| {
+        async move {
+            "<human flow>"
+            todo!()
```

Comment:
suggest using the following which will still compile if a user copy/pastes in the code snippet

```suggestion
            todo!("human flow")
```

Reply:
Thank you, updated this and added `;` after the placeholder and it compiled fine 👍